src/lib/functions.ts

```diff
@@ -142,35 +152,77 @@ export async function trigger({
       const isBroadcast =
         localServerProtocols.includes(serverProtocol) &&
         !isRemoteServer(parsedAsyncAPI, msg.server)
-      app.send(
-        new GleeMessage({
-          payload: msg.payload,
-          query: msg.query,
-          headers: msg.headers,
-          channel: msg.channel || message.channel,
-          serverName: msg.server,
-          broadcast: isBroadcast,
-        })
-      )
+      const channelName = msg.channel || message.channel
+      const operations = parsedAsyncAPI.channels().get(channelName).operations().filterBySend()
+      operations.forEach(operation => {
+        app.send(
+          new GleeMessage({
+            operation,
+            request: message,
+            payload: msg.payload,
+            query: msg.query,
+            headers: msg.headers,
+            channel: channelName,
+            serverName: msg.server,
+            broadcast: isBroadcast,
+          }))
+      })
     })
 
-    functionResult?.reply?.forEach((msg) => {
-      message.reply({
-        payload: msg.payload,
-        headers: msg.headers,
-        channel: msg.channel,
+    functionResult?.reply?.forEach((reply) => {
+      const replyMessages = createReplies(reply, message, parsedAsyncAPI)
+      console.log({ replyMessages })
+      if (!(replyMessages && replyMessages.length > 0)) {
+        return
+      }
+      replyMessages.forEach(replyMessage => {
+        const replyChannel = parsedAsyncAPI.channels().get(replyMessage.channel)
+        replyChannel.servers().forEach((server) => {
+          replyMessage.serverName = server.id()
+          app.send(
+            replyMessage
+          )
+        })
       })
+
     })
   } catch (err) {
     if (err.code === 'ERR_MODULE_NOT_FOUND') {
       const functionsPath = relative(GLEE_DIR, GLEE_FUNCTIONS_DIR)
-      const missingFile = relative(GLEE_FUNCTIONS_DIR, `${operationId}.js`)
+      const missingFile = relative(GLEE_FUNCTIONS_DIR, `${operation.id()}.js`)
       const missingPath = join(functionsPath, missingFile)
       logWarningMessage(`Missing function file ${missingPath}.`, {
         highlightedWords: [missingPath],
       })
     } else {
-      throw err
+      logError(err)
+      return
     }
   }
+}
+
+function createReplies(functionReply: GleeFunctionReturnReply, message: GleeMessage, parsedAsyncAPI: AsyncAPIDocumentInterface): GleeMessage[] {
+  const operation = message.operation
+  const reply = operation.reply()
+  if (!reply) {
+    const warningMsg = `Operation ${operation.id()} doesn't have a reply field. the return result from your function will be ignored.`
+    logWarningMessage(warningMsg)
+    return
+  }
+
+  let replyChannel = parsedAsyncAPI.channels().all().filter((c) => c.address() === reply.channel().address())[0]
```

Comment:
Please provide feedback on https://github.com/asyncapi/parser-api. This is too difficult to read. Maybe we can come up with a simpler way to get this information.

Something like:
```suggestion
  let replyChannel = parsedAsyncAPI.channels().filterByReplyAddress(reply.channel().address())[0]
```

And, IMHO, all collections should have methods like `.first()`, `.last()`, etc. We're doing a lot of `[0]` in many places. This is something that can be reported to Parser API as well. With this, it would look like this:
```suggestion
  let replyChannel = parsedAsyncAPI.channels().filterByReplyAddress(reply.channel().address()).first()
```

Reply:
it makes sense.

https://github.com/asyncapi/parser-api/issues/105
https://github.com/asyncapi/parser-api/issues/104